frontend/src/components/pages/AdminDashboard/EditCheckInOrFoodRescue.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+  Box,
+  Button,
+  CloseButton,
+  Container,
+  Flex,
+  Input,
+  Text,
+  Textarea,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hooks-helper";
+import { useHistory } from "react-router-dom";
+
+import ContentAPIClient from "../../../APIClients/ContentAPIClient";
+import * as Routes from "../../../constants/Routes";
+import { Content } from "../../../types/ContentTypes";
+
+const EditCheckInOrFoodRescue = ({
+  isCheckInView = false,
+}: {
+  isCheckInView?: boolean;
+}): React.ReactElement => {
+  const [content, setContent] = useState<Content>();
+  const [{ description, url }, setValue] = useForm({
+    description: "",
+    url: "",
+  });
+  const [interactedWith, setInteractedWith] = useState(false);
+
+  const history = useHistory();
+  const navigateToViewPage = () => {
+    history.push(
+      isCheckInView ? Routes.ADMIN_CHECK_INS : Routes.ADMIN_VIEW_DONATIONS,
+    );
+  };
+
+  useEffect(() => {
+    const getContent = async () => {
+      const contentResponse = await ContentAPIClient.getContent();
+      setContent(contentResponse);
+    };
+
+    getContent();
+  }, []);
+
+  const onSaveClick = async () => {
+    const contentResponse = await ContentAPIClient.updateContent("1", {
+      checkinDescription: String(
+        isCheckInView ? description : content?.checkinDescription,
+      ),
+      checkinUrl: String(isCheckInView ? url : content?.checkinUrl),
+      foodRescueDescription: String(
+        !isCheckInView ? description : content?.foodRescueDescription,
+      ),
+      foodRescueUrl: String(!isCheckInView ? url : content?.foodRescueUrl),
+    });
+
+    setContent(contentResponse);
+    navigateToViewPage();
+  };
+
+  return (
+    <Container
+      alignContent="left"
+      variant="calendarContainer"
+      mt={["0.5rem", "2rem"]}
+    >
+      <Box>
+        <Box display="flex" justifyContent="right">
+          <CloseButton onClick={navigateToViewPage} />
+        </Box>
+        <Text textStyle="desktopHeader4" color="black.100">{`Edit ${
+          isCheckInView ? `check in` : `food rescue`
+        } description`}</Text>
+
+        <Text
+          textStyle={["mobileHeader4", "desktopSubtitle"]}
+          color="black.100"
+          mt={["34px", "35px"]}
+        >
+          Edit description
+        </Text>
+        <Textarea
+          mt={["17px", "42px"]}
+          value={description}
+          placeholder={
+            isCheckInView
+              ? content?.checkinDescription
+              : content?.foodRescueDescription
+          }
```

Comment:
i dont think this should be placeholder, this should be infact the value. I think you should probably refactor your code to not use `useForm` imo cuz dont think its needed. You can use the value directly from content state and then in the onChange function you can update it by `setContent({...content!, whatever key ur updating depending on isCheckinView: e.target.value})`. Smth like this. This way when ur editing you are still editing the content and they are not just placeholders.

Also then in `updateContent` api call, you can just pass in `content` which will make things alot cleaner